pincer/utils/directory.py

```diff
@@ -1,14 +1,28 @@
-import os
+from __future__ import annotations
 
+from os import getcwd, chdir
+from typing import TYPE_CHECKING
 from contextlib import contextmanager
-from typing import Generator, Any
+
+if TYPE_CHECKING:
+    from typing import Generator, Any
 
 
 @contextmanager
-def chdir(path) -> Generator[str, Any, None]:
-    """Change the cwd to the given path as a context manager."""
-    current_path = os.getcwd()
-    os.chdir(path)
+def chdir(path: str) -> Generator[str, Any, None]:
+    """Change the cwd to the given path as a context manager.
+
+    Parameters
+    ----------
+    path:
+        Path to change to.
+
+    Yields
+    ------
+    Generator[:class:`str`, Any, None]
+    """
+    current_path = getcwd()
+    chdir(path)
```

Comment:
This and the `chdir` after it are supposed to be `os.chrdir`. Its been fixed on my pr but I think the maintainers should take another look at this to make sure that the code styling is right.